net/device/IotHub.cs

```diff
@@ -45,7 +56,8 @@ public async Task InitializeAsync()
 
             try
             {
-                if (Cleanup())
+                if (_deviceClient == null
+                    || Cleanup())
```

Comment:
q -> the term "cleanup" makes me think that the application is carrying out the "cleanup" stuff before shutting down; the fact that we are initializing the client instance inside this block was a bit confusing for me. Can we rename this to something like "WasClientCleanedUp"?

Reply:
I don't see it the same way, but you did make me take a new look at it. I'll make some changes.

Reply:
Updated. What do you think?

Reply:
yup, I like it